apidef/oas/security_test.go

```diff
@@ -284,3 +284,48 @@ func TestOAS_AppendSecurity(t *testing.T) {
 	})
 
 }
+
+func TestOAS_JWT(t *testing.T) {
+	const securityName = "custom"
+
+	var oas OAS
+	oas.Security = openapi3.SecurityRequirements{
+		{
+			securityName: []string{},
+		},
+	}
+
+	oas.Components.SecuritySchemes = openapi3.SecuritySchemes{
+		securityName: {
+			Value: &openapi3.SecurityScheme{
+				Type:         typeHttp,
+				Scheme:       schemeBearer,
+				BearerFormat: bearerFormatJWT,
+			},
+		},
+	}
+
+	var jwt JWT
+	Fill(t, &jwt, 0)
+	oas.Extensions = map[string]interface{}{
+		ExtensionTykAPIGateway: &XTykAPIGateway{
+			Server: Server{
+				Authentication: &Authentication{
+					SecuritySchemes: map[string]interface{}{
+						securityName: &jwt,
+					},
+				},
+			},
+		},
+	}
+
+	var api apidef.APIDefinition
+	api.AuthConfigs = make(map[string]apidef.AuthConfig)
+	oas.extractJWTTo(&api, securityName)
+
+	var convertedOAS OAS
+	convertedOAS.SetTykExtension(&XTykAPIGateway{Server: Server{Authentication: &Authentication{SecuritySchemes: map[string]interface{}{}}}})
+	convertedOAS.fillJWT(api)
+
+	assert.Equal(t, oas, convertedOAS)
```

Comment:
Can we add a test for failure, say, an invalid parameter for the security schemes? Disregard if it already exists.